components/helpers/CustomFileInput.tsx

```diff
@@ -0,0 +1,31 @@
+import { useField } from 'formik'
+import ErrorMessage from '../ErrorMessage'
+
+export default function CustomFileInput({ errors, label, ...props }) {
+  const [, , helpers] = useField(props.name)
+
+  const handleChange = (event) => {
+    const file = event.currentTarget.files[0]
+    helpers.setValue(file)
+  }
+
+  return (
+    <div>
+      <label htmlFor={props.id}>{label}</label>
+      <input
+        id={props.id}
+        name={props.name}
+        type='file'
+        onChange={handleChange}
+        className='border border-black-500 w-full'
+      />
+
+      {errors?.cover_picture &&
+        (errors.cover_picture.type ? (
+          <ErrorMessage error={String(errors.cover_picture.type)} />
+        ) : (
+          <ErrorMessage error={String(errors.cover_picture)} />
+        ))}
```

Comment:
can't you just do this?
```
{(errors?.cover_picture?.type ? (
  <ErrorMessage error={String(errors.cover_picture.type)} />
) : (
  <ErrorMessage error={String(errors.cover_picture)} />
)}
```

Reply:
no I can't, because when I open the modal the first time, it displays an "undefined" error message. Because it has 2 types of errors
`  const fileSchema = z
    .object({
      name: z.string(),
      size: z.number(),
      type: z.string().refine((value) => {
        const fileExtension = value.split('/').pop()
        return allowedImageFormats.includes(fileExtension)
      }, 'Invalid image format'),
    })
    .nullable()
    .refine((value) => value !== null, { message: 'File is required' })`

the type error, inside .object and when its empty. If I only check the type error, when starting the modal, it appears "undefined" when it's empty.